src/checkstyle/checkstyle.xml

```diff
@@ -0,0 +1,169 @@
+<?xml version="1.0"?>
+<!DOCTYPE module PUBLIC "-//Puppy Crawl//DTD Check Configuration 1.2//EN" "http://www.puppycrawl.com/dtds/configuration_1_2.dtd">
+<module name="Checker">
+
+	<module name="SuppressionFilter">
+		<property name="file" value="src/checkstyle/checkstyle-suppressions.xml" />
+	</module>
+
+	<!-- Root Checks -->
+	<module name="RegexpHeader">
+		<property name="headerFile" value="src/checkstyle/checkstyle-header.txt" />
+		<property name="fileExtensions" value="java" />
+	</module>
+	<module name="NewlineAtEndOfFile">
+		<property name="lineSeparator" value="lf"/>
+	</module>
+
+	<!-- TreeWalker Checks -->
+	<module name="TreeWalker">
+
+		<!-- Annotations -->
+ 		<module name="AnnotationUseStyle">
+ 			<property name="elementStyle" value="compact" />
+ 		</module>
+ 		<module name="MissingOverride" />
+ 		<module name="PackageAnnotation" />
+ 		<module name="AnnotationLocation">
+ 			<property name="allowSamelineSingleParameterlessAnnotation"
+ 				value="false" />
+ 		</module>
+
+		<!-- Block Checks -->
+ 		<module name="EmptyBlock">
+ 			<property name="option" value="text" />
+ 		</module>
+ 		<module name="LeftCurly" />
+ 		<module name="RightCurly">
+ 			<property name="option" value="alone" />
+ 		</module>
+ 		<module name="NeedBraces" />
+ 		<module name="AvoidNestedBlocks" />
+
+		<!-- Class Design -->
+ 		<module name="FinalClass" />
+ 		<module name="InterfaceIsType" />
+ 		<module name="HideUtilityClassConstructor" />
+ 		<module name="MutableException" />
+ 		<module name="InnerTypeLast" />
+ 		<module name="OneTopLevelClass" />
+
+		<!-- Coding -->
+ 		<module name="CovariantEquals" />
+ 		<module name="EmptyStatement" />
+ 		<module name="EqualsHashCode" />
+ 		<module name="InnerAssignment" />
+ 		<module name="SimplifyBooleanExpression" />
+ 		<module name="SimplifyBooleanReturn" />
+ 		<module name="StringLiteralEquality" />
+ 		<!--<module name="NestedForDepth">-->
+ 			<!--<property name="max" value="3" />-->
+ 		<!--</module>-->
+ 		<!--<module name="NestedIfDepth">-->
+ 			<!--<property name="max" value="3" />-->
+ 		<!--</module>-->
+ 		<!--<module name="NestedTryDepth">-->
+ 			<!--<property name="max" value="3" />-->
+ 		<!--</module>-->
```

Comment:
I'd remove these rules from the config altogether. We really have more deep `if...else` and they look good for me.
And we may have more complex structures.
WDYT?